source/deployment/certificate-based-authentication.rst

```diff
@@ -0,0 +1,174 @@
+Certificate-Based Authentication (E20)
+=======================================
+
+Certificate-based authentication (CBA) can be used identify a user or a device before granting access to Mattermost, providing an additional layer of security to access the system.
+
+Follow these steps to configure user CBA for your browser, Mattermost Desktop Apps, and the Mattermost iOS App. Before you begin, follow the `official guides to install Mattermost <https://docs.mattermost.com/guides/administrator.html#installing-mattermost>`_ on your system, including NGINX configuration as a proxy with SSL and HTTP/2, and a valid Let's Encrypt certificate.
+
+.. contents::
+  :backlinks: top
+  :local:
+  :depth: 2
+
+Run a custom build of the Mattermost server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Fork the `mm-cba-proto <https://github.com/mattermost/mattermost-server/tree/mm-cba-proto>`_ branch from the mattermost-server repository.
+2. `Use this guide <https://docs.mattermost.com/developer/dev-setup.html>`_ to create a custom build of the server based on the branch from step 1.
+
+With this custom build, the Mattermost server can be used to log in with a client side certifiate supplied as part of mutual TLS authentication from above. This guide has made some simple assumptions about how the certificates map to Mattermost users.
+
+Set up mutual TLS authentication for the Web App
```

Comment:
Generating the client certs is optional if they are going to use the ones that they already have